pkg/verifier/safe_delete.go

```diff
@@ -69,7 +70,7 @@ func BackupAndDelete(ctx context.Context, logger log.Logger, bkt, backupBkt objs
 
 	// Block uploaded, so we are ok to remove from src bucket.
 	level.Info(logger).Log("msg", "Deleting block", "id", id.String())
-	if err := block.Delete(ctx, logger, bkt, id); err != nil {
+	if err := block.MarkForDeletion(ctx, logger, bkt, id, blocksMarkedForDeletion); err != nil {
```

Comment:
Hm. This can be surprising. I think we have to rename function AND allow user to decide if to delete forcefully NOW and schedule.